src/main/python/sagemaker-reporting.py

```diff
@@ -0,0 +1,229 @@
+import json
+import boto3
+import datetime
+import os
+from aws_lambda_powertools import Logger
+
+LOG_LEVEL = os.getenv('LOG_LEVEL', 'INFO')
+LOGGER = Logger(service="SagemakerReporting", level=LOG_LEVEL)
+
+@LOGGER.inject_lambda_context
+def handler(event, context):
+    
+    emailobject = {}
+    db_client = boto3.client('dynamodb')
+    dynamodb = boto3.resource('dynamodb')
+    
+    # Get DB Table Name from environment variable
+    db_table_name = os.environ['SrvTableName']
+    email_to = os.environ['MAILTO']
+    email_cc = os.environ['MAILCC']
+    emailobject['mailfrom'] = os.environ['MAILFROM']
+    table = dynamodb.Table(db_table_name)
+    
+    # Get the AWS Account ID, Name & Region
+    env_list = {'993990991017': 'TR-LABS-PREPROD', '451191978663': 'TR-LABS-PROD'}
+    ACCOUNT_ID = context.invoked_function_arn.split(":")[4]
+    region = context.invoked_function_arn.split(":")[3]    
+    ENV = env_list[ACCOUNT_ID]
+
+    _notebookWorkspaceID = []
+    _notebookServiceOwner = []
+    _notebookInstanceStatus = []
+    _notebookInstanceName = []
+    
+    stoppedInstances = 0
+    inserviceInstances = 0
+    pendingInstances = 0
+    stoppingInstances = 0
+    failedInstances = 0
+    deletingInstances = 0
+    updatingInstances = 0
+
+    db_res = table.scan(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `scan` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `ExclusiveStartKey`, `IsTruncated`, `LastEvaluatedKey`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)